### PR TITLE
TypeScript issue on display prop

### PR DESCRIPTION
**Pull request by DilipCoder**
# Overview
#567 

I have made the union of both types here.
as here the community picker is being used so it will pick the platform-specific props and also fail-safe for other props.

  

# Test Plan
no functional changes here
once after build you can use the available props.

### CODE REVIEW

**Comment by mnajjarian** [comment]
I'm not a TypeScript expert, but I guess this also would work:

`export type ReactNativeModalDateTimePickerProps = DateTimePickerProps &
  Omit<IOSNativeProps, "value" | "mode"> |
  Omit<AndroidNativeProps, "value" | "mode">;
`
So you don't need to make extra changes.
What do you think @DilipCoder?

**Comment by DilipCoder** [comment]
> I'm not a TypeScript expert, but I guess this also would work:
> 
> `export type ReactNativeModalDateTimePickerProps = DateTimePickerProps & Omit<IOSNativeProps, "value" | "mode"> | Omit<AndroidNativeProps, "value" | "mode">; `
> 
> So you don't need to make extra changes.
> What do you think @DilipCoder?

it will do the union for all the props.  pushing the pr again

**Review by mmazzarolo — APPROVED** [review]
LGTM. I assume this is a breaking change, right? Not in the sense of introducing breaking changes at runtime, but it *may not* be retro-compatible and break type checks.

**Comment by mmazzarolo** [comment]
Also, thank you @DilipCoder and @mnajjarian :)

**Comment by DilipCoder** [comment]
> I'm not a TypeScript expert, but I guess this also would work:
> 
> `export type ReactNativeModalDateTimePickerProps = DateTimePickerProps & Omit<IOSNativeProps, "value" | "mode"> | Omit<AndroidNativeProps, "value" | "mode">; `
> 
> So you don't need to make extra changes.
> What do you think @DilipCoder?

**Comment by DilipCoder** [comment]
@mmazzarolo  can you help me to build this package locally. I want to check before it will be merged.

**Comment by mmazzarolo** [comment]
@DilipCoder there's no build step in the package. If you want to test your types, you can install the library version of this PR with something like `yarn add mmazzarolo/react-native-modal-datetime-picker#585/head` 👍 Let me know how It goes.

File: typings/index.d.ts [hunk 1]
```diff
@@ -130,7 +130,14 @@ export interface DateTimePickerProps {
    *
    * Default is 'date'
    */
-  mode?: "date" | "time" | "datetime";
+  mode?: 'date' | 'time' | 'datetime' | 'countdown';
```
Comment (mmazzarolo) [inline typings/index.d.ts:133]: We're not supporting `countdown`, could you revert this line change?

File: typings/index.d.ts [hunk 2]
```diff
@@ -130,7 +130,14 @@ export interface DateTimePickerProps {
...
+
+  /**
+   *  Display for the picker
+   * 
+   *  switches the Views for the picker
+   */
+   display?: Display | IOSDisplay
```
Comment (mmazzarolo) [inline typings/index.d.ts:140]: Can we remove the `display` definition here? It should be inferred from the changes below 👍

File: typings/index.d.ts [hunk 3]
```diff
@@ -214,7 +221,7 @@ export interface DateTimePickerProps {
 }
 
 export type ReactNativeModalDateTimePickerProps = DateTimePickerProps &
-  Omit<IOSNativeProps, "value" | "mode"> &
+  Omit<IOSNativeProps, "value" | "mode"> |
```
Comment (mmazzarolo) [inline typings/index.d.ts:224]: This seems to work 👍 
Just to make it clearer, can we split this definition in two statements?
```js
type NativePickerProps = Omit<IOSNativeProps, "value" | "mode"> | Omit<AndroidNativeProps, "value" | "mode">

export type ReactNativeModalDateTimePickerProps = DateTimePickerProps & NativePickerProps;
```

**Review by mmazzarolo — CHANGES_REQUESTED** [review]
Had some time to give it a try. There are a few things that I think we should change (just removing a few definitions) but overall looks good 👍 

The way I tested it is with the following very manual (but simple) process:

1. Create a new expo project: `expo init picker-example && cd picker-example`.
2. Install `react-native-modal-datetime-picker`: `expo install react-native-modal-datetime-picker @react-native-community/datetimepicker`.
3. Update `App.tsx` to show the picker (you can use the example from the README.md.
4. Update `picker-example/node_modules/react-native-modal-datetime-picker/typings/index.d.ts` with the changes of this PR. 
5. Play with the `display` prop in `App.tsx` and check the type errors 👍

**Comment by exneval** [comment]
Hi, thanks for the great work, just checking up, any ETA of this PR can be merged?

**Comment by mmazzarolo** [comment]
@exneval it's blocked because of the issues above ☝️ feel free to jump in if you wanna contribute 👍

**Comment by HarvWorks** [comment]
@mmazzarolo How can I help?  Can I just create a new fork to make a new PR or do I have to use @DilipCoder's fork?

**Comment by mmazzarolo** [comment]
@HarvWorks I'm fine with both options, thanks!

**Comment by mmazzarolo** [comment]
Closed #636

**Pull request closed without merging**